### DISCUSSION TITLE
Backend server for communication over web-socket?

### DISCUSSION THREAD

**Discussion by Kay2dan** [blitz-js/blitz#1400] [Q&A]
Probably you have heard this question many times, however, I too am at a point where I need to implement notification system to update user. However, I am not sure where to even begin. 

Some of my questions are:

- How would I implement a backend server purely for notifications. I imagine the system to look as such:

![image](https://user-images.githubusercontent.com/6276091/97783811-f7691900-1b91-11eb-997a-b0a4e648566f.png)

I imagine, the backend is purely for notifications and sits on the side. Otherwise, the existing way of BlitzJs will continue as it is, i.e, the front-end communites with the serverless for data.

While searching, I came across this interesting [link](https://github.com/blitz-js/blitz/discussions/1141) where ryardley, the author, discussed they initially implemented an event bus system with serverless. I would love to hear how that can be done, it is possible to have some kind of polling / web-socket implementation to notify user without a backend server?

**Comment by ryardley** [comment]
Just for clarification I have not implemented a full production ready event bus system with serverless. I am just pretty sure it is possible in theory and with some experimentation I have done. It is not desirable to take that approach for high performance workloads and the ideas in that writeup are more so that blitz could theoretically use them as a non vendor stopgap during development until apps were ready to substitute for proper systems such as sqs, rabbitmq, kafka, redpanda etc. 

So you should use a proper vendored serverless message queue and a websocket provider(Eg. Pusher).  It will be quick it will work and it will scale.

If you just want to see what is possible with serverless functions only as an experiment you can create a naive event bus using polling on the frontend to read ordered events from a db in a serverless function. I prefer portable architectures and I have written a system that calls various vercel serverless functions from an eventdispatcher function all over http so I don't need to use any proprietary systems. It worked although you have to be careful to implement security so that only your event dispatcher can call your listeners (you can use jwt to authenticate) some systems like aws allow you to colocate serverless functions in private networks so you would not need to authenticate.

**Reply by Kay2dan** [reply]
> Just for clarification I have not implemented a full production ready event bus system with serverless. I am just pretty sure it is possible in theory and with some experimentation I have done....

Thank you so much for getting back to me. 90% of what you wrote went over my head, however, I will research each of these keywords & see if I can come up with something. For me, one of the most confusing parts is the serverless deployments / infrastructure (I am conscious that that is the whole point of serverless). The positive immediate takeaway is that it is possible as I would very much like to go serverless route as much as possible & not deal with implementation of a server on the backend purely for this. Thank you Ryardley for the help.

**Reply by geoman2** [reply]
Hey folks, maybe https://github.com/serverless-stack/serverless-stack is a place to look for the solution, what do you think?
